distribution/lib/Standard/Base/0.0.0-dev/src/Data/XML/XML_Format.enso

```diff
@@ -1,25 +1,27 @@
-import Standard.Base.System.File_Format.File_Format_Metadata
-
 import project.Any.Any
 import project.Data.Text.Text
 import project.Data.XML.XML_Document
 import project.Errors.Problem_Behavior.Problem_Behavior
 import project.Network.URI.URI
 import project.Nothing.Nothing
 import project.System.File.File
-import project.System.File_Format.File_For_Read
+import project.System.File_Format_Metadata.File_Format_Metadata
 import project.System.Input_Stream.Input_Stream
 from project.Data.Text.Extensions import all
 
 ## A `File_Format` for reading and writing XML files.
 type XML_Format
     ## PRIVATE
        If the File_Format supports reading from the file, return a configured instance.
-    for_file_read : File_For_Read -> XML_Format | Nothing
-    for_file_read file:File_For_Read =
-        case file.extension of
-            ".xml" -> XML_Format
-            _ -> Nothing
+    for_read : File_Format_Metadata -> XML_Format | Nothing
+    for_read file:File_Format_Metadata =
+        content_type = file.interpret_content_type
+        from_content_type = content_type.if_not_nothing <|
+            if content_type.base_type.ends_with "/xml" then XML_Format else Nothing
```

Comment:
I think we should be more explicit here and do `application/xml` and `text/xml`.
We can also match ending with `+xml` as well.